source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -292,6 +292,10 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
         ctx.loadPrivateKey(tls_certificate.privateKey(), tls_certificate.privateKeyPath(),
                            tls_certificate.password());
       }
+
+      if (additional_init != nullptr) {
+        additional_init(ctx, tls_certificate);
```

Comment:
Instead of additionally initialize quic cert and private key by extracting them from X509 object and converting them into pem string, why not just latches the config value `tls_certificate.certificateChain()` and `tls_certificate.privateKey()` to string member variables here?

Reply:
Oh, I see. This is to support non-PEM format. But I think the conversion can be simplified. Please see the other comment.